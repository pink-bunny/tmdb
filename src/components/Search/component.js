import React from 'react';
import PropTypes from 'prop-types';
import { getIn } from 'formik';
import { Input, Typography } from 'antd';

const Search = ({
  placeholder,
  enterButton,
  field,
  onSearch,
  form: { touched, errors },
  ...props
}) => {
  const fieldTouched = getIn(touched, field.name);
  const fieldErrors = getIn(errors, field.name);

  return (
    <>
      <Input.Search
        placeholder={placeholder}
        size="large"
        enterButton={enterButton}
        onSearch={onSearch}
        onPressEnter={onSearch}
        {...field} /* eslint-disable-line */
        {...props} /* eslint-disable-line */
      />
      {fieldErrors && fieldTouched ? (
        <Typography.Paragraph
          type="danger"
          style={{ marginBottom: '16px' }}
        >
          {fieldErrors}
        </Typography.Paragraph>
      ) : null}
    </>
  );
};

Search.propTypes = {
  placeholder: PropTypes.string,
  enterButton: PropTypes.string,
  field: PropTypes.shape().isRequired,
  form: PropTypes.shape({
    touched: PropTypes.shape().isRequired,
    errors: PropTypes.shape().isRequired
  }).isRequired,
  onSearch: PropTypes.func.isRequired
};
Search.defaultProps = {
  placeholder: '',
  enterButton: 'Enter'
};

export default Search;
