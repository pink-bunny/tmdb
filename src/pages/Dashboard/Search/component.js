import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'formik';

import { Row, Col } from 'antd';
import SearchComponent from '../../../components/Search';

const Search = ({
  handleSubmit
}) => (
  <Row type="flex">
    <Col
      xs={{ span: 22, offset: 1 }}
      sm={{ span: 20, offset: 2 }}
      md={{ span: 18, offset: 3 }}
      lg={{ span: 16, offset: 4 }}
      xl={{ span: 14, offset: 5 }}
    >
      <div className="top-margin">
        <form>
          <Field
            name="search"
            placeholder="Enter movie name"
            enterButton="Search"
            onSearch={handleSubmit}
            component={SearchComponent}
          />
        </form>
      </div>
    </Col>
  </Row>
);

Search.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default Search;
