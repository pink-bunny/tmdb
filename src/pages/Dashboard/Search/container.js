import React from 'react';
import { withFormik } from 'formik';
import { connect } from 'react-redux';
import * as Yup from 'yup';

import Search from './component';

const SearchFormSchema = Yup.object().shape({
  search: Yup.string()
    .required('Search field can\'t be empty')
});

const SearchForm = withFormik({
  mapPropsToValues: () => ({ search: '' }),

  validationSchema: SearchFormSchema,

  handleSubmit: (values, { setSubmitting }) => {
    setTimeout(() => {
      alert(JSON.stringify(values, null, 2));
      setSubmitting(false);
    }, 1000);
  },

  displayName: 'SearchForm'
})(Search);

const mapDispatchToProps = {};

export default connect(null, mapDispatchToProps)(SearchForm);
