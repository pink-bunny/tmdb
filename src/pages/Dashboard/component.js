import React from 'react';
import PropTypes from 'prop-types';

import { Layout } from 'antd';
import Search from './Search';
import List from './List';

const Dashboard = ({
  list,
  loading,
  error,
  totalItems,
  fetchMovies,
  currentPage
}) => (
  <Layout>
    <Layout.Content>
      <Search />

      <div className="top-margin">
        <List
          list={list}
          error={error}
          loading={loading}
          totalItems={totalItems}
          fetchMovies={fetchMovies}
          currentPage={currentPage}
        />
      </div>
    </Layout.Content>
  </Layout>
);

Dashboard.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  totalItems: PropTypes.number.isRequired,
  fetchMovies: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired
};

export default Dashboard;
