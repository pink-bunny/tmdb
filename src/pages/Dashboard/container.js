import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Dashboard from './component';
import { isTrendingMoviesFetcing } from '../../../state/movies/selectors';
import { fetchTrendingMovies as fetchTrendingMoviesAction } from '../../../state/movies/actions';

class DashboardPage extends React.Component {
  componentDidMount() {
    const { fetchTrendingMovies } = this.props;
    fetchTrendingMovies();
  }

  render() {
    return (
      <Dashboard isTrendingMoviesFetcing />
    );
  }
}

DashboardPage.propTypes = {
  fetchTrendingMovies: PropTypes.func.isRequired
};

const mapDispatchToPtops = {
  fetchTrendingMovies: fetchTrendingMoviesAction
};

const mapStateToProps = (state) => ({
  isTrendingMoviesFetcing
});

export default connect(mapStateToProps, mapDispatchToPtops)(DashboardPage);
