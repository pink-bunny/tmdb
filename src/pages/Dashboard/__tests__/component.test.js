import React from 'react';
import { shallow } from 'enzyme';

import Dashboard from '../component';
import Search from '../Search';

describe('Dashboard component', () => {
  const requiredProps = {
    loading: false,
    error: '',
    totalItems: 0,
    fetchMovies: jest.fn(),
    currentPage: 0
  };
  it('matches snapshot with required props', () => {
    const wrapper = shallow(<Dashboard {...requiredProps} />);/* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with custom props', () => {
    const props = {
      ...requiredProps,
      list: [{
        id: 1,
        title: 'Terminator: Dark Fate',
        overview: 'Decades after Sarah Connor prevented Judgment Day.',
        poster_path: '/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg' /* eslint-disable-line */
      }]
    };
    const wrapper = shallow(<Dashboard {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot and contains Search', () => {
    const wrapper = shallow(<Dashboard {...requiredProps} />);/* eslint-disable-line */

    expect(wrapper.contains(<Search />)).toEqual(true);
  });
});
