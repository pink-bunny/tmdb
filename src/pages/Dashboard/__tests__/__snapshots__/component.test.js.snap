// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component matches snapshot with custom props 1`] = `
<Adapter>
  <Adapter>
    <Connect(WithFormik(Search)) />
    <div
      className="top-margin"
    >
      <List
        currentPage={0}
        emptyListTxt="No movies found"
        error=""
        list={
          Array [
            Object {
              "id": 1,
              "overview": "Decades after Sarah Connor prevented Judgment Day.",
              "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
              "title": "Terminator: Dark Fate",
            },
          ]
        }
        loading={false}
        onPaginationClick={[MockFunction]}
        totalItems={0}
      />
    </div>
  </Adapter>
</Adapter>
`;

exports[`Dashboard component matches snapshot with required props 1`] = `
<Adapter>
  <Adapter>
    <Connect(WithFormik(Search)) />
    <div
      className="top-margin"
    >
      <List
        currentPage={0}
        emptyListTxt="No movies found"
        error=""
        list={null}
        loading={false}
        onPaginationClick={[MockFunction]}
        totalItems={0}
      />
    </div>
  </Adapter>
</Adapter>
`;
